src/boot_O2/system_malloc.c

```diff
@@ -0,0 +1,50 @@
+#include "global.h"
+
+Arena gSystemArena;
+
+void* SystemArena_Alloc(u32 size) {
```

Comment:
```suggestion
void* SystemArena_Malloc(u32 size) {
```
I think we should sync with OoT names fully for this file.